### ISSUE TITLE
[A11Y] Pass `title` param to new iframes for screenreaders

### ISSUE THREAD

**Issue by jamesshannon** [ampproject/amphtml#12612]
Another A11Y request from a partner working through AMP.

They're trying to use `<amp-facebook-like>` and unable to set the `title` attribute for a11y purposes.

`amp-facebook-like` [calls](https://github.com/ampproject/amphtml/blob/master/extensions/amp-facebook-like/0.1/amp-facebook-like.js#L61) src/3p-frame.[getIframe()](https://github.com/ampproject/amphtml/blob/master/src/3p-frame.js#L72) which gets attributes then picks out some attributes (width and height) to add to the new iframe, but otherwise just packs them into a JSON string.

I'd propose that 3p-frame.getIframe() is modified to pass the title attribute (and maybe some other A11Y-related attributes while we're at it?). Not sure if this should be as data-* attributes or "straight".

Alternatively, amp-facebook-* should do this manually, but it seems like this should be done lower down.

**Comment by jamesshannon** [comment]
@aghassemi 

Seems like something I can tackle, but I'd like advice on:

# If it makes sense to do this automatically in getIframe(), and it's unlikely to break stuff
# If it should be `title` or `data-title`
# Any other attributes that should be added to the iframe element?

**Comment by aghassemi** [comment]
@jamesshannon Thanks for picking this up. 
I agree, it makes sense for `getIframe` to add `title` from the parent element into the `iframe`. 
I think we should go with `title` as it is semantic and already valid for all AMP and non-AMP elements in the eyes of the validator.
`getContextMetadata` probably needs to change as well to add `title` to the list of attributes it generates for `getIframe` if  there is a `title` on the element.

**Comment by jamesshannon** [comment]
The benefit of using data-title is that you then don't have two elements (the `iframe` and its parent `amp-facebook-like`) with the same title. That won't break validation, but might be confusing, especially for screen readers. But you have more knowledge in that area.

**Comment by aghassemi** [comment]
@jamesshannon it can remove `title` from the parent if having both is an issue, worth testing. But I suspect screen-reader wasn't picking up title from parent in the first place. (it may pick it up if `<amp-facebook-like>` is given a `role=button` or alike though).

**Comment by jamesshannon** [comment]
@aghassemi Ali: What are the next steps to getting this approved? Do I need to do anything special? Or is there a queue that requires patience?

**Comment by aghassemi** [comment]
@jamesshannon Reviewed and merged, thanks for the fix. Sorry for the delay, vacations and holidays... PRs usually get reviewed in within days during normal times.
